### PR TITLE
Implement JSON IO for updaters

### PR DESCRIPTION
**Pull request by trivialfis**
* Remove parameters in split evaluator.

Another small fraction of #4732 .

### CODE REVIEW

File: src/tree/split_evaluator.cc [hunk 1]
```diff
@@ -175,22 +146,6 @@ XGBOOST_REGISTER_SPLIT_EVALUATOR(ElasticNet, "elastic_net")
     return new ElasticNet(std::move(inner));
   });
 
-/*! \brief Encapsulates the parameters required by the MonotonicConstraint
-        split evaluator
-*/
-struct MonotonicConstraintParams
-    : public XGBoostParameter<MonotonicConstraintParams> {
-  std::vector<bst_int> monotone_constraints;
```
Comment (hcho3) [inline src/tree/split_evaluator.cc]: Is this field now part of TrainParam?

Comment (hcho3) [reply src/tree/split_evaluator.cc]: Indeed it is: https://github.com/dmlc/xgboost/blob/df9bdbbcb9713389759293a67e225b0985a52346/src/tree/param.h#L75

Comment (trivialfis) [reply src/tree/split_evaluator.cc]: @hcho3 Yes: https://github.com/dmlc/xgboost/blob/2dcb62ddfb2e10d58358195992df829e6d558c83/src/tree/param.h#L75

I think the one in TrainParam is older than the one in split evaluator.

Comment (hcho3) [reply src/tree/split_evaluator.cc]: Nice job deduplicating this parameter. Having one source of parameters will make our life easier.

**Comment by trivialfis** [comment]
@hcho3 

> Having one source of parameters will make our life easier.

After JSON PR, the only duplicated parameter is `max_delta_step` in `TrainParam` and `PoissonRegressionParam`.  We will be ready for some sort of parameter validation.

**Pull request merged**